src/streamlink/stream/hls.py

```diff
@@ -35,11 +36,19 @@ def __init__(self, reader, *args, **kwargs):
         kwargs["retries"] = options.get("hls-segment-attempts")
         kwargs["threads"] = options.get("hls-segment-threads")
         kwargs["timeout"] = options.get("hls-segment-timeout")
+        kwargs["ignore_names"] = options.get("hls-segment-ignore-names")
         SegmentedStreamWriter.__init__(self, reader, *args, **kwargs)
 
         self.byterange_offsets = defaultdict(int)
         self.key_data = None
         self.key_uri = None
+        if self.ignore_names:
+            # creates a regex from a list of segment names,
+            # this will be used to ignore segments.
+            self.ignore_names = list(set(self.ignore_names))
+            self.ignore_names = "|".join(list(map(re.escape, self.ignore_names)))
+            self.ignore_names_re = re.compile(r"(?:{blacklist})\.ts".format(
```

Comment:
> This will ignore every segment that ends with ...

Just a minor suggestion: you should add the dollar anchor to the end of the regex, just to be sure that a segment doesn't get matched by accident if a URL somehow contains another `.ts`, you'll never know 😃

Reply:
> ends with

i search for the **end of the filename** not for the uri end,
if there are params after `00.ts?foo=bar` it's not the end of the filename

`$` won't work for this

---

if i would parse the uri and just work with the _path_, then sure
but using `urlparse` might be to advanced.

Reply:
True... I had static segment URLs in mind without query data.
`\.ts(?:?.*)$` should work, but we could also just leave it as it is, it's good enough. :)